File: backend/ReptiRealm/src/ReptiRealm/Models/Reptile.cs
using System.ComponentModel.DataAnnotations;

namespace ReptiRealm.Models
{
    public class Reptile : BaseEntity
    {
        [Required]
        public required string Name { get; set; }
        public char? Sex { get; set; }
        public DateOnly? DOB { get; set; }
        public virtual Species? Species { get; set; }
        public virtual ICollection<Shed>? Sheds { get; set; }
        public virtual ICollection<Weight>? Weights { get; set; }
        public virtual ICollection<Defecation>? Defecations { get; set; }
        public virtual ICollection<Feed>? Feeds { get; set; }
        public virtual ICollection<Morph>? Morphs { get; set; }

    }
}

File: App/lib/models/reptile.dart
import 'package:reptirealm/models/feed.dart';
import 'package:reptirealm/models/morph.dart';
import 'package:reptirealm/models/shed.dart';
import 'package:reptirealm/models/species.dart';
import 'package:reptirealm/models/weight.dart';
import 'package:reptirealm/models/defecation.dart';


class Reptile {

  // REPTILE INFORMATION
  final String? id;
  final String name;
  final String? sex;
  final DateTime? dob;
  final Species? species;
  final List<Morph>? morphs;

  // REPTILE DATA
  final List<Feed>? feeds;
  final List<Shed>? sheds;
  final List<Defecation>? defecations;
  final List<Weight>? weights;

  Reptile({
    this.id,
    required this.name,
    this.sex,
    this.dob,
    this.species,
    this.morphs,

    this.feeds,
    this.sheds,
    this.defecations,
    this.weights,
  });
}
